src/AdminGuesser.js

```diff
@@ -71,11 +91,76 @@ AdminGuesser.defaultProps = {
       },
     },
   }),
-  appLayout: Layout,
+
+  resourceSchemaAnalyzer,
+
+  children,
+  ...rest
+}) => {
+  const [resources, setResources] = useState();
+  const [fetching, setFetching] = useState(true);
+  const [error, setError] = useState();
+
+  if (appLayout) {
+    console.warn(
+      'You are using deprecated prop "appLayout", it was replaced by "layout", see https://github.com/marmelab/react-admin/issues/2918',
+    );
+  }
+  if (loginPage === true && process.env.NODE_ENV !== 'production') {
```

Comment:
why warn only in production here and not for the other two warnings?

Reply:
Makes sense but FYI it's how it's done in react-admin: https://github.com/marmelab/react-admin/blob/56b1d4096a5e68e442d29b1b7b12699d3bfc3f4b/packages/react-admin/src/Admin.tsx#L107-L121.

Reply:
Then it's worth an issue in the react-admin repo!

Reply:
Or a PR :slightly_smiling_face: https://github.com/marmelab/react-admin/pull/4279